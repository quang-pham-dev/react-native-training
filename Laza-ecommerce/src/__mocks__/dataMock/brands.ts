/* A data array. */

export const brand = {
  id: '0',
  name: 'Nike',
  logoUrl:
    'https://toppng.com/uploads/preview/ike-logo-png-hd-nike-logo-png-image-free-download-nike-logo-sv-11562865034fkhqkcoas6.png'
};

export const brands = [
  {
    id: '0',
    name: 'Nike',
    logoUrl:
      'https://toppng.com/uploads/preview/ike-logo-png-hd-nike-logo-png-image-free-download-nike-logo-sv-11562865034fkhqkcoas6.png'
  },
  {
    id: '1',
    name: 'Adidas',
    logoUrl:
      'https://www.pikpng.com/pngl/m/42-429866_adidas-logo-png-images-free-download-rh-pngimg.png'
  },
  {
    id: '2',
    name: 'Fila',
    logoUrl:
      'https://toppng.com/uploads/preview/fila-logo-vector-download-free-11574216707jcmvfu58fy.png'
  },
  {
    id: '3',
    name: 'Puma',
    logoUrl:
      'https://banner2.cleanpng.com/20180613/gsj/kisspng-tracksuit-puma-logo-brand-clothing-puma-logo-5b20dc6b232426.487874081528880235144.jpg'
  },
  {
    id: '4',
    name: 'Gucci',
    logoUrl: 'https://esquilo.io/png/thumb/XBwczcMncYeff65-Vector-Gucci-Logo-PNG-Free-Download.png'
  }
];
