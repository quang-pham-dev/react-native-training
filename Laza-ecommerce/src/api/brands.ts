// API
import http from 'api/http';

// Constants
import { ENDPOINTS } from 'constants/Common';
import { PRODUCT_PAGINATION } from 'constants/Products';

// Types
import { IBrand, IBrandsResponse } from 'types/models/Brands';

// get all brands
const getBrands = async (limit: number): Promise<IBrandsResponse> => {
  return await http.get(
    `${ENDPOINTS.BRAND.GET_BRANDS}?_page=${PRODUCT_PAGINATION.PAGE}&_limit=${limit}`,
  );
};

// get brand by id
const getBrandById = async (id: string): Promise<IBrand> => {
  return await http.get(`${ENDPOINTS.BRAND.GET_BRANDS}?id=${id}`);
};
export const brandsService = {
  getBrands,
  getBrandById,
};
