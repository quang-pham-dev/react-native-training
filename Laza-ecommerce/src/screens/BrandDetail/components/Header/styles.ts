import { Platform, StyleSheet } from 'react-native';

// Themes
import Colors from 'themes/Colors';
import Metrics from 'themes/Metrics';

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    width: '100%',
    marginTop: Platform.OS === 'android' ? 40 : 45,
    position: 'absolute',
    backgroundColor: Colors.primaryBackground
  },

  brandLogoWrapper: {
    height: 45,
    width: 70,
    padding: Metrics.padding.sm,
    borderRadius: 10,
    backgroundColor: Colors.secondaryBackground
  },

  brandLogo: {
    width: '100%',
    height: '100%',
    resizeMode: 'contain'
  }
});

export default styles;
