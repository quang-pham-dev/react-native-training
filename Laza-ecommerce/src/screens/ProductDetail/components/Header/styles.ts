import { Platform, StyleSheet } from 'react-native';

// Themes
import Colors from 'themes/Colors';
import Fonts from 'themes/Fonts';
import Metrics from 'themes/Metrics';
import { height } from 'themes/Dimensions';

const styles = StyleSheet.create({
  headerContainer: {
    height: height * 0.5,
    backgroundColor: Colors.lightGray,
    alignItems: 'center'
  },

  actionsWrapper: {
    position: 'absolute',
    width: '100%',
    marginTop: Platform.OS === 'android' ? 40 : 45,
    zIndex: 1,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    paddingHorizontal: Metrics.padding.lg
  },

  headerImageWrapper: {
    width: '100%',
    justifyContent: 'center',
    alignItems: 'center'
  },

  image: {
    width: '100%',
    height: '100%',
    resizeMode: 'cover'
  }
});

export default styles;
