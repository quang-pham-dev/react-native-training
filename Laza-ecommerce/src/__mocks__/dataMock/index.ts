import { user } from './user';
import { brand, brands } from './brands';
import { product, products } from './products';

export { brand, brands, product, products, user };
