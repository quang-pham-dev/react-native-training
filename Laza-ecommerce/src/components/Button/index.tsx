import React from 'react';
import { Image, Text, TouchableOpacity } from 'react-native';
// Types
import { ButtonProps } from 'types/Button';
// Styles
import styles from './styles';

const Button = ({
  onPress = () => {},
  text,
  textStyles,
  buttonStyles,
  icon,
  iconStyles,
  disabled,
  children,
  ...props
}: ButtonProps) => {
  return (
    <TouchableOpacity
      style={[styles.button, buttonStyles]}
      {...props}
      onPress={onPress}
      disabled={disabled}
      activeOpacity={0.8}
    >
      {icon && <Image style={[styles.icon, iconStyles]} source={icon} />}
      {text && <Text style={[styles.text , textStyles]}>{text}</Text>}
      {children}
    </TouchableOpacity>
  );
};

export default Button;
