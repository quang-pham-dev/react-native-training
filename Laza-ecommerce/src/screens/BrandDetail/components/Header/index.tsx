import React, { memo, useCallback } from 'react';
import { Image, ImageBackground, Pressable, View } from 'react-native';

// LIBS
import isEqual from 'react-fast-compare';

// Components
import Button from 'components/Button';

// Theme
import IMAGES from 'themes/Images';

// Style
import styles from './styles';

type HeaderProps = {
  navigation: any;
  logoUrl: any;
};

const Header = ({ logoUrl, navigation }: HeaderProps) => {
  // handle back button
  const handlePressBackIcon = useCallback(() => {
    navigation.goBack();
  }, []);

  return (
    <View style={styles.container}>
      <Button icon={IMAGES.iconBack} onPress={handlePressBackIcon} type='Circle' />
      <View style={styles.brandLogoWrapper}>
        {Boolean(logoUrl) && <Image style={styles.brandLogo} source={{ uri: logoUrl }} />}
      </View>
      <Button icon={IMAGES.iconCart} type='Circle' />
    </View>
  );
};

export default memo(Header, isEqual);
