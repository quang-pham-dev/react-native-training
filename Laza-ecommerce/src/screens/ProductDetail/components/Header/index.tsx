import React, { useCallback } from 'react';
import { ImageBackground, View } from 'react-native';

// Components
import Button from 'components/Button';

// Theme
import IMAGES from 'themes/Images';

// Types
import { IProductDetailHeaderProps } from 'types/screens/ProductDetail';

// Style
import styles from './styles';

const Header = ({ source, navigation }: IProductDetailHeaderProps) => {
  // handle back button
  const handlePressBack = useCallback(() => {
    navigation.goBack();
  }, []);

  return (
    <View style={styles.headerContainer}>
      <View style={styles.actionsWrapper}>
        <Button icon={IMAGES.iconBack} onPress={handlePressBack} type='Circle' />
        <Button icon={IMAGES.iconCart} type='Circle' />
      </View>
      {/* End action header */}

      <View style={styles.headerImageWrapper}>
        {Boolean(source) && <ImageBackground style={styles.image} source={{ uri: source }} />}
      </View>
      {/* End Header block */}
    </View>
  );
};

export default Header;
