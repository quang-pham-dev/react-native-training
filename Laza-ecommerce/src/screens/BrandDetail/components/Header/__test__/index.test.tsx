import React from 'react';

// LIBS
import renderer from 'react-test-renderer';

// Components
import Button from 'components/Button';
import Header from 'screens/BrandDetail/components/Header';

// Mock data
import { brand } from '__mocks__/dataMock/brands';

describe('Brand detail Header', () => {
  let props = {
    logoUrl: brand.logoUrl,
    navigation: {
      navigate: jest.fn(),
      goBack: jest.fn()
    }
  };

  const tree = renderer.create(<Header {...props} />);

  test('should render correctly', () => {
    const component = tree.toJSON();
    expect(component).toMatchSnapshot();
  });

  test('should call function handlePressBackIcon', () => {
    const backButton = tree.root.findAllByType(Button.type)[0];
    backButton.props.onPress();

    expect(props.navigation.goBack).toHaveBeenCalled();
  });
});
