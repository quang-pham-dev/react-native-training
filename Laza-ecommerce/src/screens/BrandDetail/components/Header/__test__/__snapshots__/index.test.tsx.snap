// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Brand detail Header should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginTop": 45,
      "position": "absolute",
      "width": "100%",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "height": 45,
        "opacity": 1,
        "width": 45,
      }
    }
  >
    <Image
      source={
        Object {
          "process": [Function],
        }
      }
      style={Object {}}
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#F5F6FA",
        "borderRadius": 10,
        "height": 45,
        "padding": 10,
        "width": 70,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://toppng.com/uploads/preview/ike-logo-png-hd-nike-logo-png-image-free-download-nike-logo-sv-11562865034fkhqkcoas6.png",
        }
      }
      style={
        Object {
          "height": "100%",
          "resizeMode": "contain",
          "width": "100%",
        }
      }
    />
  </View>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "height": 45,
        "opacity": 1,
        "width": 45,
      }
    }
  >
    <Image
      source={
        Object {
          "process": [Function],
        }
      }
      style={Object {}}
    />
  </View>
</View>
`;
